packages/jest-snapshot/src/__tests__/utils.test.js

```diff
@@ -195,12 +195,74 @@ test('serialize handles \\r\\n', () => {
 
 describe('DeepMerge', () => {
   it('Correctly merges objects with property matchers', () => {
-    const target = {data: {bar: 'bar', foo: 'foo'}};
-    const matcher = expect.any(String);
-    const propertyMatchers = {data: {foo: matcher}};
+    /* eslint-disable sort-keys */
+    // to keep keys in numerical order rather than alphabetical
+    const target = {
+      data: {
+        one: 'one',
+        two: 'two',
+        three: [
+          {
+            four: 'four',
+            five: 'five',
+          },
+          // Include an array element not present in the propertyMatchers
```

Comment:
I added an extra array element to ensure behavior between differently-sized arrays works as expected